rust/src/measurements/rappor/mod.rs

```diff
@@ -0,0 +1,111 @@
+use crate::error::Fallible;
+use crate::metrics::DiscreteDistance;
+use crate::measures::MaxDivergence;
+use crate::domains::{AtomDomain, VectorDomain};
+use crate::core::{Function, Measurement, PrivacyMap};
+use crate::traits::{InfSub, InfDiv, InfLn, InfMul, samplers::SampleBernoulli};
+
+
+/// Make a Measurement that implements Basic RAPPOR
+/// 
+/// # Arguments
+/// * `f` - Per-bit flipping probability. Must be in $(0, 1]$.
+/// * `m` - number of ones set in each boolean vector (1 if one-hot encoding, more if using a bloom filter)
+///
+/// See paper for more details: https://arxiv.org/abs/1407.6981
+///
+/// eps = ln((2-f)/f) 2
+pub fn make_rappor(
+    input_domain: VectorDomain<AtomDomain<bool>>,
+    input_metric: DiscreteDistance,
+    f: f64,
+    m: i32,
+    constant_time: bool
+) -> Fallible<Measurement<VectorDomain<AtomDomain<bool>>, Vec<bool>, DiscreteDistance, MaxDivergence<f64>>> {
+    if input_domain.size.is_none() {
+        return fallible!(MakeMeasurement, "RAPPOR requires a known number of categories.")
+    }
+
+    if !(0.0..=1.0).contains(&f) {
+        return fallible!(MakeMeasurement, "f must be in (0, 1].")
```

Comment:
```suggestion
        return fallible!(MakeMeasurement, "f must be in (0, 1]")
```

(I won't flag the rest of these below.)